store/sqlstore/user_store.go

```diff
@@ -456,119 +457,59 @@ func applyRoleFilter(query sq.SelectBuilder, role string, isPostgreSQL bool) sq.
 	return query.Where("u.Roles LIKE ? ESCAPE '*'", roleParam)
 }
 
-func applyMultiRoleFilters(query sq.SelectBuilder, roles []string, teamRoles []string, channelRoles []string) sq.SelectBuilder {
-	queryString := ""
-	if len(roles) > 0 && roles[0] != "" {
-		schemeGuest := false
-		schemeAdmin := false
-		schemeUser := false
+func applyMultiRoleFilters(query sq.SelectBuilder, roles []string, teamRoles []string, channelRoles []string, isPostgreSQL bool) sq.SelectBuilder {
+	sqOr := sq.Or{}
 
+	if len(roles) > 0 && roles[0] != "" {
```

Comment:
Why are does `roles` sometimes have an empty string in the first position?

Reply:
it doesnt, _unless_ someone someone were to pass an empty string as the value for roles... I just decided to protect against that case, do you think its overkill?